recipe/patches/0006-remove-third-party-references.patch

```diff
@@ -0,0 +1,70 @@
+diff --git a/lib/Target/LLVMIR/LLVMIRTranslation.cpp b/lib/Target/LLVMIR/LLVMIRTranslation.cpp
+index cfb79868d2..2f63d0d108 100644
+--- a/lib/Target/LLVMIR/LLVMIRTranslation.cpp
++++ b/lib/Target/LLVMIR/LLVMIRTranslation.cpp
+@@ -118,6 +118,12 @@ static std::map<std::string, std::string> getExternLibs(mlir::ModuleOp module) {
+ 
+   if (!funcs.empty()) {
+     static const std::string libdevice = "libdevice";
++    // first search for environmental path
++    std::string env_path = ::triton::tools::getenv("TRITON_LIBDEVICE_PATH");
++    if (!env_path.empty()) {
++      externLibs.try_emplace(libdevice, env_path);
++      return externLibs;
++    }
+     namespace fs = std::filesystem;
+     // Search for libdevice relative to its library path if used from Python
+     // Then native code is in `triton/_C/libtriton.so` and libdevice in
+
```

Comment:
> PS: This change [...] is taken from upstream and will land in the next triton version.

Could you please separate the patches, and add some attribution / references to where the respective bits come from?

Best is to do `git format-patch <base>` after a chain of respective commits (cherry-picked/rebased or self-authored).

Reply:
Done.